sql/catalyst/src/main/java/org/apache/spark/sql/connector/expressions/Transform.java

```diff
@@ -34,11 +34,6 @@
    */
   String name();
 
-  /**
-   * Returns all field references in the transform arguments.
-   */
-  NamedReference[] references();
```

Comment:
We need to provide a default implementation here, which is exactly the same with `GeneralScalarExpression`